modules/core/04-channel/v2/types/genesis.go

```diff
@@ -0,0 +1,134 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+
+	channeltypesv1 "github.com/cosmos/ibc-go/v9/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v9/modules/core/24-host"
+)
+
+// NewPacketState creates a new PacketState instance.
+func NewPacketState(channelID string, sequence uint64, data []byte) PacketState {
+	return PacketState{
+		ChannelId: channelID,
+		Sequence:  sequence,
+		Data:      data,
+	}
+}
+
+// Validate performs basic validation of fields returning an error upon any failure.
+func (ps PacketState) Validate() error {
+	if ps.Data == nil {
+		return errors.New("data bytes cannot be nil")
+	}
+	return validateGenFields(ps.ChannelId, ps.Sequence)
+}
+
+// NewPacketSequence creates a new PacketSequences instance.
+func NewPacketSequence(channelID string, sequence uint64) PacketSequence {
+	return PacketSequence{
+		ChannelId: channelID,
+		Sequence:  sequence,
+	}
+}
+
+// Validate performs basic validation of fields returning an error upon any failure.
+func (ps PacketSequence) Validate() error {
+	return validateGenFields(ps.ChannelId, ps.Sequence)
+}
+
+// NewGenesisState creates a GenesisState instance.
+func NewGenesisState(
+	channels []IdentifiedChannel, acks, receipts, commitments []PacketState,
+	sendSeqs []PacketSequence, nextChannelSequence uint64,
+) GenesisState {
+	return GenesisState{
+		Channels:            channels,
+		Acknowledgements:    acks,
+		Receipts:            receipts,
+		Commitments:         commitments,
+		SendSequences:       sendSeqs,
+		NextChannelSequence: nextChannelSequence,
```

Comment:
will basically be duplicated for v1/v2. Need it here too tho in case some chain only uses eureka channels.